lib/braintreehttp/encoder.js

```diff
@@ -3,13 +3,14 @@
 let Json = require('./serializer/json').Json;
 let Text = require('./serializer/text').Text;
 let Multipart = require('./serializer/multipart').Multipart;
+let MultipartRelated = require('./serializer/multipart_related').MultipartRelated;
 let FormEncoded = require('./serializer/form_encoded').FormEncoded;
 let zlib = require('zlib');
 
 class Encoder {
 
   constructor() {
-    this._encoders = [new Json(), new Text(), new Multipart(), new FormEncoded()];
+    this._encoders = [new Json(), new Text(), new MultipartRelated(), new Multipart(), new FormEncoded()];
```

Comment:
It seems like there's a lot of code duplication between the `multipart` and `mutlipart_related`. Is there any way we can generalize some of the logic or even make one work for both content types?